### PR TITLE
Create testcoverage.rst

### PR DESCRIPTION
**Pull request by Blue-Hope**
related to https://github.com/behave/behave-django/issues/119

### CODE REVIEW

File: docs/testcoverage.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+.. code-block:: bash
+
+    $ python -m http.server --directory ./cov
+    
+Warning for behave-django
+--------------------
+
+Internally, the time before_all is executed seems to be later than the time when django loads the modules set in each app.
+
+So sometimes it is necessary to reload django app's modules for accurate test coverage measurement.
```
Comment (bittner) [inline docs/testcoverage.rst:52]: Whoa! Reloading modules. That looks hacky. We should find a proper solution for that problem!

We have [django_ready](https://behave-django.readthedocs.io/en/stable/setup.html) already. The `before_all` should run before that, but it seems that [we actually give Django precedence](https://github.com/behave/behave-django/blob/master/behave_django/environment.py#L114-L115). That might need to be changed. A PR is welcome!

File: docs/testcoverage.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
+Test Coverage
+====================
```
Comment (bittner) [inline docs/testcoverage.rst:2]: Please make the underlining exactly as long as the title text.

**Review by bittner — CHANGES_REQUESTED** [review]

File: docs/testcoverage.rst (hunk 1 above)
Comment (Blue-Hope) [reply docs/testcoverage.rst:52]: Okay, I'll check the hook.

File: docs/testcoverage.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+At First, you should install Coverage.py dependency
+
+.. code-block:: bash
+
+    $ pip install coverage
+
+Environment.py
+--------------
+
+In environment.py , write below code on function before_all to start measuring test coverage
```
Comment (bittner) [inline docs/testcoverage.rst:18]:
```suggestion
In ``environment.py``, add the code snippet below in the ``before_all`` function to start measuring test coverage:
```

**Review by bittner — CHANGES_REQUESTED** [review]
Can you also add the testcoverage chapter to the documentation [index](https://github.com/behave/behave-django/blob/master/docs/index.rst), please?

Then, I think, this PR can be merged. :checkered_flag:

**Comment by Blue-Hope** [comment]
Okay

**Comment by Blue-Hope** [comment]
@bittner 
All review requests have been updated except for the module reloading part. We will fix the reload part to use django_ready later.

File: docs/testcoverage.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+In ``environment.py``, add the code snippet below in the ``before_all`` function to start measuring test coverage:
+
+.. code-block:: python
+
+    import coverage
+
+    def before_all(context):
+        cov = coverage.Coverage()
+        cov.start()
+        context.cov = cov
```
Comment (bittner) [inline docs/testcoverage.rst:27]: This looks like code that we could add [to our management command](https://github.com/behave/behave-django/blob/main/behave_django/management/commands/behave.py#L151-L153), doesn't it? – The feature could be activated by a `--coverage` command line switch.

File: docs/testcoverage.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Then write below code to end up measuring test coverage.
+You can save the coverage result on html format.
+
+.. code-block:: python
+
+    def after_all(context):
+        cov = context.cov
+        cov.stop()
+        cov.save()
+        cov.html_report(directory="./cov")
```
Comment (bittner) [inline docs/testcoverage.rst:38]: Same here: Couldn't we simply integrate that code [in our management command](https://github.com/behave/behave-django/blob/main/behave_django/management/commands/behave.py#L156-L157) and allow our users to activate the feature via a command line switch?

Then, the coverage report should probably not be written to an explicit location. The developer using the feature might want to control this via [coverage settings](https://coverage.readthedocs.io/en/latest/config.html) from the outside.

File: docs/testcoverage.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+          members = inspect.getmembers(app)
+          modules = map(
+              lambda keyval: keyval[1],
+              filter(lambda keyval: inspect.ismodule(keyval[1]), members),
+          )
+          for module in modules:
+              try:
+                  importlib.reload(module)
+              except:
+                  continue
```
Comment (bittner) [inline docs/testcoverage.rst:75]: This code could be placed in a [coverage module](https://github.com/behave/behave-django/blob/main/behave_django/coverage.py) (`behave_django/coverage.py`) and triggered by _behave-django_'s management command, when needed, as suggested above.

Then, there would be no need to do copy+paste from the documentation to get coverage analysis running with your BDD tests. – What do you think?

**Pull request closed without merging**